src/Makefile

```diff
@@ -121,10 +125,14 @@ GHERKIN_CLI_OBJS= \
 	$(MKDIR_CMD) $(dir $@)
 	$(AR) $(AR_FLAGS) $@ $(UTILITIES_OBJS) $(PARSER_OBJS) $(AST_OBJS) $(COMPILER_OBJS) $(PICKLES_OBJS) $(EVENT_OBJS)
 
+../libs/libgherkin.so: $(UTILITIES_OBJS) $(PARSER_OBJS) $(AST_OBJS) $(COMPILER_OBJS) $(PICKLES_OBJS) $(EVENT_OBJS)
+	$(MKDIR_CMD) $(dir $@)
+	$(CC) $(GCC_SO_FLAGS) -o ../libs/libgherkin.so.1.0 $(UTILITIES_OBJS) $(PARSER_OBJS) $(AST_OBJS) $(COMPILER_OBJS) $(PICKLES_OBJS) $(EVENT_OBJS)
+
 ../bin/gherkin_generate_tokens$(EXT): ../libs/libgherkin.a $(GENERATE_TOKEN_OBJS) Makefile
 	$(MKDIR_CMD) $(dir $@)
 	$(LD) $(LD_FLAGS) $(GENERATE_TOKEN_OBJS) -L../libs/ -lgherkin $(LD_LIBS) -o $@
 
-../bin/gherkin$(EXT): ../libs/libgherkin.a $(GHERKIN_CLI_OBJS) Makefile
+../bin/gherkin$(EXT): ../libs/libgherkin.a ../libs/libgherkin.so $(GHERKIN_CLI_OBJS) Makefile
```

Comment:
"../libs/libgherkin.so" is not used for building "../bin/gherkin$(EXT)", so it should not be a dependency of the make rule.

Reply:
Again, I do not understand.  ../libs/libgherkin.so does build its target because it calls CC to build the actual shared object so I do not know what to do to address your comment.

Reply:
The "../bin/gherkin" binary is statically linking the gherkin library. It does so using by using the "../libs/libgherkin.a" archive file. When you execute `make ../bin/gherkin`, "../libs/libgherkin.so" should not be built because it is not needed/used to produce "../bin/gherkin".

I think you added this to get "../libs/libgherkin.so" build when running `make` (with no argument), but this is the wrong way to achieve that (introducing a fake dependency to a make rule). To achieve that "../libs/libgherkin.so" build when running `make`, the "all" rule should be changed.